libcudacxx/include/cuda/std/__cmath/exponential_functions.h

```diff
@@ -142,166 +146,114 @@ template <class _Integer, enable_if_t<_CCCL_TRAIT(is_integral, _Integer), int> =
 
 // frexp
 
-#if _CCCL_CHECK_BUILTIN(builtin_frexp) || _CCCL_COMPILER(GCC)
-#  define _CCCL_BUILTIN_FREXPF(...) __builtin_frexpf(__VA_ARGS__)
-#  define _CCCL_BUILTIN_FREXP(...)  __builtin_frexp(__VA_ARGS__)
-#  define _CCCL_BUILTIN_FREXPL(...) __builtin_frexpl(__VA_ARGS__)
-#endif // _CCCL_CHECK_BUILTIN(builtin_frexp)
-
-// clang-cuda fails with fatal error: error in backend: Undefined external symbol "frexp"
-#if _CCCL_CUDA_COMPILER(CLANG)
-#  undef _CCCL_BUILTIN_FREXPF
-#  undef _CCCL_BUILTIN_FREXP
-#  undef _CCCL_BUILTIN_FREXPL
-#endif // _CCCL_CUDA_COMPILER(CLANG)
-
-[[nodiscard]] _CCCL_API inline float frexp(float __x, int* __e) noexcept
+template <class _Tp>
+[[nodiscard]] _CCCL_API inline constexpr _Tp __frexp_impl(_Tp __x, int* __exp) noexcept
```

Comment:
```suggestion
[[nodiscard]] _CCCL_API constexpr _Tp __frexp_impl(_Tp __x, int* __exp) noexcept
```
`constexpr` functions are implicitly `inline`, applies to the whole PR